airflow_dbt_python/hooks/dbt.py

```diff
@@ -555,22 +599,101 @@ class DbtSnowflakeHook(DbtHook):
     conn_type = "snowflake"
     hook_name = "dbt Snowflake Hook"
     conn_params = [
-        DbtConnectionParam("conn_type", "type", "postgres"),
+        DbtConnectionParam("conn_type", "type", conn_type),
         "host",
         "schema",
-        DbtConnectionParam("login", "user"),
-        "password",
+        DbtConnectionParam(
+            "login",
+            "user",
+            depends_on=lambda x: x.extra_dejson.get("authenticator", "") != "oauth",
```

Comment:
I feel like we could just have a utility function that we can re-use here instead of using lambda:

```python
def make_extra_dejson_compare_callable(key, default, comparison_operator, expected):
    def compare_extra_dejson_value(conn):
        return comparison_operator(conn.extra_dejson.get(key, default), expected))
        
    return compare_extra_dejson_value
```

Then here:
```python
import operator
...
        DbtConnectionParam(
            "login",
            "user",
            depends_on=make_extra_dejson_compare_callable("authenticator", "", operator.ne, "oauth"),
        ),
```

This one is kind of a nitpick. I do prefer an approach like this to using lambda, but I think the other comments are more important.

Reply:
I liked this idea and decided to combine recurring parameters together